modules/tinymce/src/themes/silver/main/ts/ui/dialog/ImagePanel.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { AlloyComponent, Behaviour, Memento, Representing, SimpleSpec } from '@ephox/alloy';
+import { Dialog } from '@ephox/bridge';
+import { Optional, Singleton } from '@ephox/katamari';
+import { Attribute, Css, Height, SugarElement, Width } from '@ephox/sugar';
+
+import { ComposingConfigs } from '../alien/ComposingConfigs';
+
+type ImagePanelSpec = Omit<Dialog.ImagePanel, 'type'>;
+
+export interface ImagePanelData {
+  url: string;
+  zoom: Optional<number>;
+}
+
+// Does this belong somewhere else?
+const loadImage = (image: SugarElement<HTMLImageElement>): Promise<SugarElement<HTMLImageElement>> => new Promise((resolve) => {
+  const loaded = () => {
+    image.dom.removeEventListener('load', loaded);
+    resolve(image);
+  };
+
+  if (image.dom.complete) {
+    resolve(image);
+  } else {
+    image.dom.addEventListener('load', loaded);
```

Comment:
As a side note... it looks like this is also largely duplicated in core, though has more fixes/case handling than this: https://github.com/tinymce/tinymce/blob/3865a8cb3420afde40926a62680059d046cf28cb/modules/imagetools/src/main/ts/ephox/imagetools/util/Conversions.ts#L16

Reply:
That code is part of the library we're removing - but I suspected while developing this that the whole "wait for image to load" concept should move to Sugar or something like that. So maybe it needs to be moved and the library then uses it (that will be super complex to coordinate, though).

I left it without an error handler because dialogs don't have an error management system, as far as I can tell, so I'm not sure what should happen if there is an error loading this image long after the dialog has been displayed 🤔

Reply:
I guess in this case if it errors we just want to reject the promise and then do a noop in the dialog, as it'll just mean the image won't render? It's either that or we'd have to show a notification or something.

Reply:
It was already a no-op in the dialog, the image just never rendered, but now it does that with a console log (and the browser probably logs about the rejected promise as well).